connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/ActivatedExecutableManager.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.executable;
+
+import io.camunda.connector.api.error.ConnectorRetryException;
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorFactory;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorReportingContext;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ActivatedExecutableManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ActivatedExecutableManager.class);
+  private final InboundConnectorFactory inboundConnectorFactory;
+  private final InboundConnectorReportingContext context;
+  private final ScheduledExecutorService reactivationExecutor;
+  boolean isCancelled;
+  private InboundConnectorExecutable<InboundConnectorContext> executable;
+
+  public ActivatedExecutableManager(
+      InboundConnectorFactory inboundConnectorFactory,
+      InboundConnectorReportingContext inboundConnectorContext,
+      InboundConnectorExecutable<InboundConnectorContext> executable,
+      ScheduledExecutorService reactivateScheduler) {
+    this.inboundConnectorFactory = inboundConnectorFactory;
+    this.context = inboundConnectorContext;
+    this.executable = executable;
+    this.reactivationExecutor = reactivateScheduler;
+    this.isCancelled = true;
+    this.context.loadCancellationCallback(this::cancel);
+  }
+
+  public InboundConnectorReportingContext context() {
+    return context;
+  }
+
+  public boolean isCancelled() {
+    return isCancelled;
+  }
+
+  public void cancel(Throwable throwable) {
+    switch (throwable) {
+      case ConnectorRetryException connectorRetryException -> {
+        handleCancellation(throwable);
+        handleRestart(connectorRetryException);
+      }
+      default -> handleCancellation(throwable);
+    }
+  }
+
+  private void handleRestart(ConnectorRetryException connectorRetryException) {
```

Comment:
maybe could we rename it to `activateWithDelay` or similar.
I also realize that `ConnectorRetryException` doesn't allow us to provide `errorVariables` that are defined in the parent `ConnectorException`. That's on me, I missed it a few month ago. You can ignore it if you want to.